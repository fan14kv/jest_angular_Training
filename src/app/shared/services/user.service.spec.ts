import { TestBed } from "@angular/core/testing"
import { UserService } from "./user.service"
import { Directive } from "@angular/core";
import { UserInterface } from "../types/user.interface";
import { HttpClientTestingModule, HttpTestingController } from "@angular/common/http/testing";

describe("UserService",()=> {
    let userService : UserService;
    let httpClient : HttpTestingController;
    beforeEach(()=>{
        TestBed.configureTestingModule({
            imports: [HttpClientTestingModule],
            providers:[UserService]
        });
        userService = TestBed.inject(UserService);
        httpClient = TestBed.inject(HttpTestingController);
    });

    it("Service Should be called",()=>{
        expect(userService).toBeTruthy();
    });

    describe('addUser',()=>{
        it('should add a user',()=>{
            const user:UserInterface={
                id:"3",
                name:"Raja",
                isActive: true
            };
            expect(user).toMatchSnapshot();
            userService.addUser(user);
            expect(userService.users$.getValue()).toEqual([{id:"3",name:"Raja",isActive: true}])
        });

        it('Reactive Subscription Test',()=>{
            const user:UserInterface={
                id:"3",
                name:"kumar",
                isActive: true
            }
            userService.addUser(user);
            userService.users$.subscribe(user =>{
                expect(user).toEqual([{id:"3",name:"kumar",isActive: true}]);
            })
        });
    });
    describe('removeUser',()=>{
        it("should remove a user",()=>{
            userService.users$.next([{
                id:"1",
                name:"raj",
                isActive: true
            },
            {
                id:"3",
                name:"kumar",
                isActive: true
            }
        ]);
            userService.removeUser("3");
            expect(userService.users$.getValue()).toEqual([{id:"1",name:"raj",isActive: true}])
        });
    });
});