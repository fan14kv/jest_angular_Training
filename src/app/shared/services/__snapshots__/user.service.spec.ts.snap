// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService addUser should add a user 1`] = `
{
  "id": "3",
  "name": "Raja",
}
`;
